app/views/pages/release-notes/2020/MARCH/16.liquid

```diff
@@ -0,0 +1,93 @@
+
+* allow to filter GraphQL Query `users` by `related_model`
+
+Example query will find all company owners (users) whose company's (which is a model in his example) property "codename" is either "ES" or "FI":
+```
+query {
+  users(
+    per_page: 10
+    filter: {
+      related_models: {
+        join: { join_on_property: "id" foreign_property: "company_owner_id" }
+        filter: {
+          model_schema_name: { value: "company" }
+          properties: [{ name: "codename" value_in: ["ES", "FI"] }]
+        }
+      }
+    }
+  ) {
+    results {
+      id
+      email
+      companies: related_models(join_on_property: "id", foreign_property: "company_owner_id" model_schema_name: "company"){
+        codename: property(name: "codename")
+      }
+    }
+  }
+}
+```
+
+* allow to filter GraphQL Query  `users` by `related_users`
+
+Example query will find all users whose boss's first name is John:
+```
+query {
+  users(
+    per_page: 10
+    filter: {
+      related_users: {
+        join: { join_on_property: "boss_id" foreign_property: "id" }
+        filter: {
+          first_name: { value: "John" }
+        }
+      }
+    }
+  ) {
+    results {
+      external_id
+      email
+      boss: related_users(join_on_property: "boss_id", foreign_property: "id") {
+        external_id
+        first_name
+        email
+      }
+    }
+  }
+}
+```
+
+<h4 class="release-note release-note__improved">IMPROVED</h4>
+
+* deploy/sync based on `physical_file_path` for all admin resources
+
+Behind the scenes `pos-cli` was relying on platformOS to find admin resources like forms, graphql queries, model schemas based on different properties, usually name. This has been improved and from now on physical_file_path will be used. This allowed us to add two enhancements to `pos-cli` -  `pos-cli pull`, which would be the reverse of deploy - it will create `app/` and `modules/` direcctory based on current DB state (using the newest naming, without unnecessary default values) and `pos-cli sync` will also listen to file deletion and act accordingly. Those changes will be released separately, you can follow them at [pos-cli Changelog](https://github.com/mdyd-dev/pos-cli/blob/master/CHANGELOG.md).
+
+* Respect directories in GraphQL file names
+
+Previously example file app/graphql/my_diectory/my_query.graphql was invokable via {% raw %}{% graphql res = 'my_query' %}{% endraw %}. It was the only resource which behaved like this. After the change, the expected way to invoke it is {% raw %}{% graphql res = 'my_directory/my_query' %}{% endraw %}. For backwards compatibility, the previous way still works, however a deprecation log is generated. For this particular example, it would look like this:
```

Comment:
```suggestion
Previously example file `app/graphql/my_diectory/my_query.graphql` was invokable via {% raw %}{% graphql res = 'my_query' %}{% endraw %}. It was the only resource which behaved like this. After the change, the expected way to invoke it is {% raw %}{% graphql res = 'my_directory/my_query' %}{% endraw %}. For backwards compatibility, the previous way still works, however a deprecation log is generated. For this particular example, it would look like this:
```

Reply:
I would add a sentence why we changed that - apart from inconsistency, it also allows cleaner structures, not like this: https://github.com/mdyd-dev/todo-app/tree/master/app/graphql